### PR TITLE
Add ignorable private flag to driver to match changes in frontend

### PR DESCRIPTION
**Pull request by elsh**
Resolves rdar://107638447

Frontend changes: https://github.com/apple/swift/pull/65336

### CODE REVIEW

**Comment by elsh** [comment]
@swift-ci test

**Review by artemcm — APPROVED** [review]

**Comment by elsh** [comment]
@swift-ci test

File: Sources/makeOptions/makeOptions.cpp [hunk 1]
```diff
@@ -64,6 +64,7 @@ enum SwiftFlags {
   SwiftAPIDigesterOption = (1 << 17),
   NewDriverOnlyOption = (1 << 18),
   ModuleInterfaceOptionIgnorable = (1 << 19),
+  ModuleInterfaceOptionIgnorablePrivate = (1 << 20),
```
[the remark this answers is not in the document]
Comment (elsh) [reply Sources/makeOptions/makeOptions.cpp:67]: done

**Comment by elsh** [comment]
@swift-ci test

**Comment by elsh** [comment]
@swift-ci test

**Comment by elsh** [comment]
@swift-ci test Windows

**Pull request merged**